fractal.sdk/src/main/java/com/yanchware/fractal/sdk/domain/livesystem/LiveSystemAggregate.java

```diff
@@ -146,6 +160,21 @@ public Collection<String> validate() {
     }
 
     public void delete() throws InstantiatorException {
+        ensureEnvironmentExists(environment.id(), String.format("delete LiveSystem [id: '%s']", id));
         service.deleteLiveSystem(getId().toString());
     }
+
+    private void ensureEnvironmentExists(EnvironmentIdValue environmentId, String action) throws EnvironmentException, EnvironmentNotFoundException {
+        try {
+            var existingEnvironment = environmentService.tryGetById(environmentId);
+            if (existingEnvironment == null) {
+                throw new EnvironmentNotFoundException( // Use a more specific exception type
```

Comment:
Why the comment here? I don't think we need a more specific exception, but If you want it, let's create it.